weather-station/background/main.js

```diff
@@ -0,0 +1,358 @@
+      updateWappstoData({
+        accessToken: response.data.access_token,
+        refreshToken: response.data.refresh_token,
+        expiresIn: response.data.expires_in
+      });
+
+      getStationData();
+    })
+    .catch(function(error) {
+      console.log(error);
+    });
+};
+
+// Send request to refresh token and update tokens with new values
+let getRefreshToken = function() {
+  let refreshToken = data.get("refreshToken");
+
+  axios({
+    method: "POST",
+    headers: {
+      Host: "api.netatmo.com",
+      "Content-type": "application/x-www-form-urlencoded;charset=UTF-8"
+    },
+    url: "/oauth2/token",
+    baseURL: "https://api.netatmo.com/",
+    data: querystring.stringify({
+      grant_type: "refresh_token",
+      refresh_token: refreshToken,
+      client_id: config.clientId,
+      client_secret: config.clientSecret
+    })
+  })
+    .then(function(response) {
+      updateWappstoData({
+        accessToken: response.data.access_token,
+        refreshToken: response.data.refresh_token,
+        expiresIn: response.data.expires_in
+      });
+
+      getStationData();
+    })
+    .catch(function(error) {
+      console.log(error);
+    });
+};
+
+// Use device data to create device, values and state and then add device to the network
+let addDevicesToNetwork = function(deviceData) {
+  deviceData.forEach(function(device) {
+    let deviceToAdd = createDevice(device);
+
+    if (deviceToAdd) {
+      let deviceDataTypes = device.data_type;
+
+      deviceDataTypes.forEach(function(dataType) {
+        let valueToAdd = createValue(dataType, device);
+
+        deviceToAdd.get("value").push(valueToAdd);
+      });
+      network.get("device").push(deviceToAdd);
+    }
+  });
+};
+
+// Get the users station data
+let getStationData = function() {
+  let accessToken = data.get("accessToken");
+
+  if (!accessToken) {
+    getAccessToken();
+  }
+
+  axios({
+    method: "GET",
+    headers: {
+      Host: "api.netatmo.com",
+      Authorization: "Bearer " + accessToken
+    },
+    url: "/getstationsdata",
+    baseURL: "https://api.netatmo.com/api/",
+    data: querystring.stringify({
+      device_id: config.deviceId,
+      get_favorites: false
+    })
+  })
+    .then(function(response) {
+      // Data of the Main Module - every station has this device
+      let deviceData = response.data.body.devices;
+      // Saving station name to display in the FG
+      let stationName = deviceData[0].station_name;
+
+      if (data.get("stationName") !== stationName) {
+        updateWappstoData({ stationName: stationName });
+      }
+
+      addDevicesToNetwork(deviceData);
+      // Data of the modules associated with the Main Module
+      let moduleData = response.data.body.devices[0].modules;
+
+      addDevicesToNetwork(moduleData);
+      // Saving network
+      saveNetwork();
+      // Save unreachable devices if any
+      if (unreacheableDevices.length > 0) {
+        updateWappstoData({

```

Comment:
You could save unreachable devices under the device(create a custom value for it)